src/libraries/Native/Unix/System.Globalization.Native/pal_timeZoneInfo.c

```diff
@@ -223,7 +224,7 @@ static void FixupTimeZoneGenericDisplayName(const char* locale, const UChar* tim
         }
 
         // Make a UChar[] version of the test time zone id for use in the API calls.
-        u_uastrcpy(testTimeZoneId, testId);
+        u_uastrncpy(testTimeZoneId, testId, (int32_t)strlen(testId));
```

Comment:
This should be `TZID_LENGTH`, not `(int32_t)strlen(testId)`. (It's the size of the destination, not the size of the source.)

Reply:
That make sense.